vpr/src/device/rr_spatial_lookup.h

```diff
@@ -89,7 +90,32 @@ class RRSpatialLookup {
                   int ptc,
                   e_side side);
 
-    /* TODO: Add an API remove_node() to unregister a node from the look-up */
+    /* Mirror the last dimension of a look-up, i.e., a list of nodes, from a source coordinate to 
+     * a destination coordinate.
+     * This function is mostly need by SOURCE and SINK nodes which are indexable in multiple locations.
+     * Considering a bounding box (x, y)->(x + width, y + height) of a multi-height and multi-width grid, 
+     * SOURCE and SINK nodes are indexable in any location inside the boundry.
+     *
+     * Note: currently this function only accept SOURCE/SINK nodes. May unlock for other depending on needs
```

Comment:
Yes. This function will copy a vector of node indices from a source ``[src_x][src_y][type][side]`` to a destination ``[des_x][des_y][type][side]``. It is required when building the lookup for SOURCE and SINK nodes, which are supposed to be indexable at any location inside a large block ``(x,y)->(x+width, y+height)``.

I agree that we should have a high-level API to do the job, while this low-level API can be a private one. Here are my thoughts:
- We can create a high-level API like ``expand_nodes()`` which will copy the vector from a source coordinate to all the coordinates in a bounding box.
- I also think that it is better to rework the ``find_node()`` API so that we always search the lowest (x,y) corner when dealing with large blocks. But this may require the data structure to be dependent on ``DeviceGrid`` information (it needs to identify if  a grid has height > 1 as well as width >1)

We can keep brainstorming on this front. I have left comments in the header file as a TODO item.

I have also added comments (a quick example) about how to use the API.